src/json/pull_parser.cr

```diff
@@ -507,7 +507,7 @@ class JSON::PullParser
   end
 
   private def expect_kind(kind)
-    parse_exception "Expected #{kind} but was #{@kind}" unless @kind == kind
+    parse_exception "Expected #{kind} but was #{@kind} for attribute: #{@string_value}" unless @kind == kind
```

Comment:
This implementation is too weak. For example this result is your expected?

```crystal
require "json"

class Foo
  JSON.mapping({
    foo: Int32,
  })

  def initialize(@foo)
  end
end

Foo.from_json(%{{"foo": "bar"}}) # raises: Expected int but was string for attribute: bar at 1:14 (JSON::ParseException)
```

I think you can add `attribute_name` argument to `expect_kind` and each `read_xxx` method,  then use it to show attribute name inside `JSON.mapping`.